src/main/java/org/cryptomator/ui/unlock/UnlockWorkflow.java

```diff
@@ -107,6 +113,16 @@ protected void succeeded() {
 			case ASK -> Platform.runLater(() -> {
 				window.setScene(successScene.get());
 				window.show();
+				double x = mainWindow.getX() + (mainWindow.getWidth() - window.getWidth()) / 2;
+				double y = mainWindow.getY() + (mainWindow.getHeight() - window.getHeight()) / 2;
+				if(!mainWindow.isShowing()) {
+					Screen screen = Screen.getScreensForRectangle(mainWindow.getX(), mainWindow.getY(), mainWindow.getWidth(), mainWindow.getHeight()).get(0);
```

Comment:
Can't `Screen.getScreensForRectangle(...)` return an empty list, leading to an IOOB exception?

How about simply using `Screen.getPrimary()`?